crates/nu-cli/src/commands/str_/substring.rs

```diff
@@ -344,19 +327,20 @@ mod tests {
             expectation("andr", (0, 4)),
             expectation("andre", (0, 5)),
             expectation("andres", (0, 6)),
-            expectation("andres", (0, -6)),
-            expectation("ndres", (0, -5)),
-            expectation("dres", (0, -4)),
-            expectation("res", (0, -3)),
-            expectation("es", (0, -2)),
-            expectation("s", (0, -1)),
+            expectation("", (0, -6)),
+            expectation("a", (0, -5)),
+            expectation("an", (0, -4)),
+            expectation("and", (0, -3)),
+            expectation("andr", (0, -2)),
+            expectation("andre", (0, -1)),
```

Comment:
This makes more sense to me 👍 

Something for us to think about will be how we can do an inclusive end. I'd love to be able to do something like:

```
str substring 5,-1
```

to mean "start at index 5, all the way and including the end". Another option is to interpret a missing value as an inclusive end:

```
str substring 5,
```

Not for this PR, of course 🙂

Reply:
```
echo 'good nushell' | str substring 5, -> nushell
```
The above behaviour already exists. We set the default value of a missing `end` value with `isize::max_value()`.

Reply:
Ahhh, you're totally right. I should've continued past these lines https://github.com/defstryker/nushell/blob/8e62dde93c4bcb3294ac9783b6082d5f3293d9a8/crates/nu-cli/src/commands/str_/substring.rs#L229-L248 😅